File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m = len(matrix)
        n = len(matrix[0])
        for row in matrix:
            if target < row[0]:
                return False
            if target > row[n-1]:
                continue
            for val in row:
                if val == target:
                    return True
                else:
                    continue
        return False
            
        

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] mat, int target) {
        int m = mat.length;
        int n = mat[0].length;
        int x = m - 1;
        int y = 0 ;
        while (x >= 0 && y < n){
            if (mat[x][y] > target){
                x --;
            } else if (mat[x][y] < target){
                y ++;
            } else {
                return true;
            }
        }
        return false;
    }
}